packages/cells/src/cells/date-picker-cell.tsx

```diff
@@ -1,15 +1,80 @@
-import * as React from "react";
-import { CustomCell, CustomRenderer, drawTextCell, GridCellKind } from "@glideapps/glide-data-grid";
+import React from "react";
+import { assertNever } from "./common/support"
+import {
+    CustomCell,
+    CustomRenderer,
+    drawTextCell,
+    GridCellKind,
+    ProvideEditorCallback,
+} from "@glideapps/glide-data-grid";
 
 interface DatePickerCellProps {
     readonly kind: "date-picker-cell";
     readonly date: Date | undefined;
     readonly displayDate: string;
-    readonly format: "date" | "datetime-local";
+    readonly dateKind: DateType;
 }
 
+export type DateType = "date" | "time" | "datetime-local";
+
+export const formatValueForHTMLInput = (dateKind: DateType, date: Date): string => {
+    switch (dateKind) {
+        case "date":
+            return date.toISOString().split("T")[0];
+        case "datetime-local":
+            return date.toISOString().replace("Z", "");
+        case "time":
+            return date.toISOString().split("T")[1].replace("Z", "");
+        default:
+            assertNever(dateKind);
+    }
+};
+
 export type DatePickerCell = CustomCell<DatePickerCellProps>;
 
+const Editor: ReturnType<ProvideEditorCallback<DatePickerCell>> = cell => {
+    const cellData = cell.value.data;
+    const { date, displayDate, dateKind } = cellData;
+    const cellDataDate =
+        cellData.date ??
+        // now as date with adjusted timezone to allow for displayDate to be correct
+        new Date(new Date().getTime() - new Date().getTimezoneOffset() * 60000);
+    const value = formatValueForHTMLInput(dateKind, cellDataDate);
+
+    return (
+        <input
+            required
+            style={{ minHeight: 26, border: "none", outline: "none" }}
+            type={dateKind}
+            value={value}
+            autoFocus={true}
+            onChange={event => {
+                // handle when clear is clicked and value has been wiped
+                if (event.target.value === "") {
+                    cell.onChange({
```

Comment:
Why did you get rid of the try catch now?

Reply:
Added it back! This is what we're looking for?

Reply:
Also, I will have to head out soon for a good amount of time but I might come back to this later tonight / tomorrow morning :) so if I stop replying / commiting, sorry but we can continue later :)
Thanks for the fast replies and commentary.

Reply:
Hey no problem, looks great so far!